### ISSUE TITLE
Please use aria-hidden for codeblock <a> tag for accessibility

### ISSUE THREAD

**Issue by jooyoungseo** [jgm/pandoc#6352]
Hello, I am a blind person using text-to-speech screen reading software.

I am filing an accessibility issue found in the current `pandoc` v2.9.2.1.

It seems like `pandoc` adds `<a>` tag to each code line for html output like below.

``` html
<a href="#cb1-1"></a>
```

Since this is an empty `<a>` tag, screen readers read aloud the value of `href` attribute for every single code line.

For example, [strong emphasis to small caps lua filter](https://pandoc.org/lua-filters.html#cb1-1) should be seen like below for sighted people:

``` lua
return {
  {
    Strong = function (elem)
      return pandoc.SmallCaps(elem.c)
    end,
  }
}
```

But, the code above is heard to me like below via screen reader:

```
#cb1-1
 return {
#cb1-2
  {
#cb1-3
  Strong = function (elem)
#cb1-4
  return pandoc.SmallCaps(elem.c)
#cb1-5
  end,
#cb1-6
  }
#cb1-7
 }
```

This is not ideal.

To avoid this confusion for screen reader, please consider adding [`aria-hidden = "true"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-hidden_attribute) attribute to each `<a>` tag like below:

``` html
<a href="#cb1-1" aria-hidden = "true"></a>
```

I have tested and confirmed that screen readers skip the annoying `<a>` tags when this technique is used.

**Comment by jgm** [comment]
Thanks for reporting this. Do the span elements used to highlight code also cause problems, or does the screen reader just ignore them?  One possibility is always using `--no-highlight`, which would remove all of these things.  But I suppose you've also got to read pandoc-generated documents that you didn't produce yourself.

In any case, it shouldn't be hard to add the aria-hidden attribute. It would need to be done in jgm/skyligting-core, which defines the way our highlighted code is formatted.

**Comment by jgm** [comment]
One more question about this.  When the `numberLines` class is used, pandoc's highlighting includes a line number within this 'a' element. When line numbers are not shown, we just include an empty 'a' element (as in your example) so we have the anchor.  I see why it woud be desirable to have `aria-hidden` set in the case where it's empty. But what about when it encloses a line number?  Is it better in that case not to set `aria-hidden="true"`, or should it still be set?  (That would mean, I assume, that the line number wouldn't be read out.)

**Comment by jooyoungseo** [comment]
Thank you very much for your quick and kind response, @jgm!

1. `<span>` element itself does not cause any redundant info; only empty `<a>` element is the main issue for screen readers in this case.
2. I do not think we need `aria-hidden = "true"` when `numberLines` class is used as it puts each number as an element content which is correctly read by screen readers. Yes, screen readers will read aloud each line number before the actual code; however, that is what's meant to be for readers. Thus, I think the information should be delivered as equally as possible.

**Comment by jgm** [comment]
This has been added.

**Comment by jooyoungseo** [comment]
@jgm, I was attempting to create a `lua` filter to address this issue. As the fix will be likely to be implemented in `pandoc` > v2.9.2.1; I needed a workaround for older versions.

I know I have to use `CodeBlock`; however, how would you find an empty `<a>` tag under `CodeBlock`? Typically, `<a>` tag should be presented as `Link` in AST, but I was not able to find that this time.

Any starting point would be greatly appreciated.

**Comment by jgm** [comment]
You can't really do this in a filter (not without great difficulty anyway), since the highlighting is added in the writer, and the filter operates prior to the writer.

I'd suggest postprocessing the output instead.
